src/Service.Tests/SqlTests/RestApiTests/RestBodyFlexibleModeTests/RestBodyNonStrictModeTests.cs

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Net;
+using System.Threading.Tasks;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Azure.DataApiBuilder.Service.Tests.SqlTests.RestApiTests
+{
+    /// <summary>
+    /// Class containing integration tests to validate scenarios when we operate in non-strict mode for REST request body,
+    /// i.e. we allow extraneous fields to be present in the request body.
+    /// </summary>
+    public abstract class RestBodyNonStrictModeTests : RestApiTestBase
+    {
+        #region Positive tests
+
+        /// <summary>
+        /// Test to validate that extraneous fields are allowed in request body when we operate in runtime.rest.request-body-strict = false.
```

Comment:
Would be good to note that this single test method covers PUT, PATCH, POST requests and also uniquely tests that values in the URI for PK's are honored instead of any differing PK values included in the request body.

Reply:
done